planner/implementation/datasource.go

```diff
@@ -171,3 +171,82 @@ func NewIndexScanImpl(scan *plannercore.PhysicalIndexScan, tblColHists *statisti
 		tblColHists: tblColHists,
 	}
 }
+
+// IndexLookUpReaderImpl is the Implementation of PhysicalIndexLookUpReader.
+type IndexLookUpReaderImpl struct {
+	baseImpl
+
+	KeepOrder   bool
+	tblColHists *statistics.HistColl
+}
+
+// NewIndexLookUpReaderImpl creates a new table reader Implementation.
+func NewIndexLookUpReaderImpl(reader plannercore.PhysicalPlan, hists *statistics.HistColl) *IndexLookUpReaderImpl {
+	base := baseImpl{plan: reader}
```

Comment:
This is wired. `reader` might be a `Projection` here. 

Can we keep the `Projection` in another field, and set it as the plan in `AttachChildren`? like:
```golang
type IndexLookUpReaderImpl struct {
	baseImpl

	KeepOrder   bool
        ExtraProjection *plannercore.Projection
	tblColHists *statistics.HistColl
}
func (impl *..) AttachChildren(children) {
    reader := impl.plan.(*plannercore.PhysicalIndexLoopUpReader)
    if impl.ExtraProjection != nil {
        impl.plan = impl.ExtraProjection
    }
    ...

}
```

Reply:
Good advice.

Reply:
Done.